pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/ActivityInstanceDao.java

```diff
@@ -240,6 +255,18 @@ default void saveSubstitutions(long instanceId, Map<String, String> substitution
         DBUtils.checkInsert(variables.size(), inserted.length);
     }
 
+    @SqlQuery("select distinct sub.variable_name"
```

Comment:
I've looked at the query plan for these queries and they seem fairly performant -- a large part of the query leverages indices in the tables. There are some operations -- like the `distinct` and `group by` -- that adds some weight to the queries, but overall seems fine.

Reply:
Once per study per export per activity. Not a big deal.